connectors/serial/src/main/java/quarks/connectors/serial/SerialDevice.java

```diff
@@ -0,0 +1,49 @@
+/*
+# Licensed Materials - Property of IBM
+# Copyright IBM Corp. 2015,2016 
+*/
+package quarks.connectors.serial;
+
+
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.util.concurrent.TimeUnit;
+
+import quarks.function.BiConsumer;
+import quarks.function.BiFunction;
+import quarks.topology.TStream;
+import quarks.topology.TopologyElement;
+
+/**
+ * Generic interface for a serial port.
+ *
+ */
+public interface SerialDevice extends TopologyElement {
+	
+	/**
+	 * Set the initialization function for this port.
+	 * Can be used to send setup instructions to the
+	 * device connected to this serial port.
+	 * <BR>
+	 * {@code initializer.accept(out, in)} is called once, passing the output
+	 * and input streams for this port.
+	 * 
+	 * @param initializer Function to be called when the application runs.
+	 */
+	public void setInitializer(BiConsumer<OutputStream,InputStream> initializer);
```

Comment:
It's early monday so I may just be a bit groggy...
I'm not understanding why it's desirable to have OutputStream and InputStream show up in this interface.  i.e., why isn't that info just a private aspect of the supplied ``initializer`` and ``driver``?  ``Runnable initializer``, ``Supplier<T> driver``?
Also I imagine that it's common to have interrupt driven devices.  That would seem to imply an eventual need for a ``pubic <T> TStream<T> events(...)``.  Maybe there will also be an eventual desire for a ``generate(Supplier<T>)`` like method?  At a high level, I'm missing the value of ~duplicating one or more stream-source methods present in ``Topology``.